### ISSUE TITLE
pip install fails

### ISSUE THREAD

**Issue by johentsch** [DCMLab/wavescapes#15]
Wanted to check if the matplotlib warnings from #14 also occur in fresh environments with Python 3.9 and 3.10. Both times, upon `pip install wavescpaes` this happened (environment, here, is called `ms9`).

```
    ERROR: Command errored out with exit status 1:
     command: ~/anaconda3/envs/ms9/bin/python -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-w7vliao8/madmom_560942881f574f65bc7fd5b98854071e/setup.py'"'"'; __file__='"'"'/tmp/pip-install-w7vliao8/madmom_560942881f574f65bc7fd5b98854071e/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-pip-egg-info-hzezozcr
         cwd: /tmp/pip-install-w7vliao8/madmom_560942881f574f65bc7fd5b98854071e/
    Complete output (5 lines):
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-install-w7vliao8/madmom_560942881f574f65bc7fd5b98854071e/setup.py", line 12, in <module>
        from Cython.Build import cythonize, build_ext
    ModuleNotFoundError: No module named 'Cython'
    ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/c7/a3/9f3de3e8068a3606331134d96b84c8db4f7624d6715be8ab3c1f56e6731d/madmom-0.16.1.tar.gz#sha256=64a86a29106e7c4e0c5f4ec96801c2d1a8492db710e4fe27e097da5517d68cb2 (from https://pypi.org/simple/madmom/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
  Using cached madmom-0.16.tar.gz (20.0 MB)
    ERROR: Command errored out with exit status 1:
     command: ~/anaconda3/envs/ms9/bin/python -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-w7vliao8/madmom_a3705a9be54344499274024a2a35a1f8/setup.py'"'"'; __file__='"'"'/tmp/pip-install-w7vliao8/madmom_a3705a9be54344499274024a2a35a1f8/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-pip-egg-info-v_ttujlq
         cwd: /tmp/pip-install-w7vliao8/madmom_a3705a9be54344499274024a2a35a1f8/
    Complete output (5 lines):
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-install-w7vliao8/madmom_a3705a9be54344499274024a2a35a1f8/setup.py", line 12, in <module>
        from Cython.Build import cythonize, build_ext
    ModuleNotFoundError: No module named 'Cython'
    ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/cb/fd/72afb59ac0dc760df0abe7f60a8474e727d3cca96ac3b44b56b26d5e983e/madmom-0.16.tar.gz#sha256=7f63f18bfb5e54ba7ea59ad4e1713f489fd5409c9f1aaaf706757be6e4d9b042 (from https://pypi.org/simple/madmom/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
  Using cached madmom-0.15.1.tar.gz (15.7 MB)
    ERROR: Command errored out with exit status 1:
     command: ~/anaconda3/envs/ms9/bin/python -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-w7vliao8/madmom_6878236cefcd4ae5994d39d6b47e8897/setup.py'"'"'; __file__='"'"'/tmp/pip-install-w7vliao8/madmom_6878236cefcd4ae5994d39d6b47e8897/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-pip-egg-info-402cj980
         cwd: /tmp/pip-install-w7vliao8/madmom_6878236cefcd4ae5994d39d6b47e8897/
    Complete output (5 lines):
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-install-w7vliao8/madmom_6878236cefcd4ae5994d39d6b47e8897/setup.py", line 12, in <module>
        from Cython.Build import cythonize, build_ext
    ModuleNotFoundError: No module named 'Cython'
    ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/b7/bb/b9b88d504f5ed3d87dd8a952c4feb37165a96ba1015f912cd4d6f050d75f/madmom-0.15.1.tar.gz#sha256=2a1325b0c0932611cb8da3890cae6320540fba806881a9dc7957aa909568acee (from https://pypi.org/simple/madmom/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
  Using cached madmom-0.15.tar.gz (15.7 MB)
    ERROR: Command errored out with exit status 1:
     command: ~/anaconda3/envs/ms9/bin/python -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-w7vliao8/madmom_dbf6f775ab724e9399aee7fcf0b815b3/setup.py'"'"'; __file__='"'"'/tmp/pip-install-w7vliao8/madmom_dbf6f775ab724e9399aee7fcf0b815b3/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-pip-egg-info-_f0xao9q
         cwd: /tmp/pip-install-w7vliao8/madmom_dbf6f775ab724e9399aee7fcf0b815b3/
    Complete output (5 lines):
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-install-w7vliao8/madmom_dbf6f775ab724e9399aee7fcf0b815b3/setup.py", line 12, in <module>
        from Cython.Build import cythonize, build_ext
    ModuleNotFoundError: No module named 'Cython'
    ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/fb/cb/67f1a1e5a82e520d60899973e3ea0a96125a662ea2c7bd90a9147970fd60/madmom-0.15.tar.gz#sha256=8ad1e68be070a26095ec8c972b859a47c3cc86b51aebdcf672a39878c0e4af41 (from https://pypi.org/simple/madmom/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
  Using cached madmom-0.14.1.tar.gz (14.3 MB)
    ERROR: Command errored out with exit status 1:
     command: ~/anaconda3/envs/ms9/bin/python -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-w7vliao8/madmom_85620b86b2c84b20b6a615d0e6de1a5e/setup.py'"'"'; __file__='"'"'/tmp/pip-install-w7vliao8/madmom_85620b86b2c84b20b6a615d0e6de1a5e/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-pip-egg-info-0grmdctt
         cwd: /tmp/pip-install-w7vliao8/madmom_85620b86b2c84b20b6a615d0e6de1a5e/
    Complete output (5 lines):
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-install-w7vliao8/madmom_85620b86b2c84b20b6a615d0e6de1a5e/setup.py", line 11, in <module>
        from Cython.Distutils import build_ext
    ModuleNotFoundError: No module named 'Cython'
    ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/14/06/0b9d9d1b271171e0d8c739856f51a7553dacf662e4a5753fc7c54655144f/madmom-0.14.1.tar.gz#sha256=1a6b8a736e561d5adabd49f0946e531f2439a9d1af4dd5452b8188f4da97b170 (from https://pypi.org/simple/madmom/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
  Using cached madmom-0.14.tar.gz (16.3 MB)
    ERROR: Command errored out with exit status 1:
     command: ~/anaconda3/envs/ms9/bin/python -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-w7vliao8/madmom_b880bd3c38c04dd7ad757ba732b24351/setup.py'"'"'; __file__='"'"'/tmp/pip-install-w7vliao8/madmom_b880bd3c38c04dd7ad757ba732b24351/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-pip-egg-info-acvnvk8l
         cwd: /tmp/pip-install-w7vliao8/madmom_b880bd3c38c04dd7ad757ba732b24351/
    Complete output (5 lines):
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-install-w7vliao8/madmom_b880bd3c38c04dd7ad757ba732b24351/setup.py", line 11, in <module>
        from Cython.Distutils import build_ext
    ModuleNotFoundError: No module named 'Cython'
    ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/8f/46/64dd93a1211e710a7b75a92c1fb223d601d391744f1c0829d7e7afe6791f/madmom-0.14.tar.gz#sha256=e0dcb65aba5fb8816a6792a44aca02f356b9003943079a5a9863d12fb2863d0a (from https://pypi.org/simple/madmom/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
  Using cached madmom-0.13.2.tar.gz (3.8 MB)
    ERROR: Command errored out with exit status 1:
     command: ~/anaconda3/envs/ms9/bin/python -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-w7vliao8/madmom_396b4267275844abb0e3382abf5b70e9/setup.py'"'"'; __file__='"'"'/tmp/pip-install-w7vliao8/madmom_396b4267275844abb0e3382abf5b70e9/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-pip-egg-info-46w_mtje
         cwd: /tmp/pip-install-w7vliao8/madmom_396b4267275844abb0e3382abf5b70e9/
    Complete output (5 lines):
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-install-w7vliao8/madmom_396b4267275844abb0e3382abf5b70e9/setup.py", line 11, in <module>
        from Cython.Distutils import build_ext
    ModuleNotFoundError: No module named 'Cython'
    ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/6a/c1/3a93ec4ce0be0b0f68409d5974137f9999a833976cceae971d876ff86a04/madmom-0.13.2.tar.gz#sha256=aa8fcd858bb29f016f425792df655f40040098052b065c0fa0a5393a6a70d171 (from https://pypi.org/simple/madmom/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
  Using cached madmom-0.13.1.tar.gz (3.8 MB)
    ERROR: Command errored out with exit status 1:
     command: ~/anaconda3/envs/ms9/bin/python -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-w7vliao8/madmom_f892464228f74816a2e00126de69ca70/setup.py'"'"'; __file__='"'"'/tmp/pip-install-w7vliao8/madmom_f892464228f74816a2e00126de69ca70/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-pip-egg-info-tct5w13e
         cwd: /tmp/pip-install-w7vliao8/madmom_f892464228f74816a2e00126de69ca70/
    Complete output (5 lines):
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-install-w7vliao8/madmom_f892464228f74816a2e00126de69ca70/setup.py", line 11, in <module>
        from Cython.Distutils import build_ext
    ModuleNotFoundError: No module named 'Cython'
    ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/bb/5e/c2ef8193a8201c6fbc6b10fa5e44ea9e50a6a7643590de6fbf893809ecb9/madmom-0.13.1.tar.gz#sha256=0a553d17fc70d481718a640eb342667410e2ca8af5cec43e5459ab036b330bda (from https://pypi.org/simple/madmom/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
  Using cached madmom-0.13.tar.gz (3.8 MB)
    ERROR: Command errored out with exit status 1:
     command: ~/anaconda3/envs/ms9/bin/python -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-w7vliao8/madmom_82f62a15469b470e85d6112c19a5cd64/setup.py'"'"'; __file__='"'"'/tmp/pip-install-w7vliao8/madmom_82f62a15469b470e85d6112c19a5cd64/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-pip-egg-info-nhfri75z
         cwd: /tmp/pip-install-w7vliao8/madmom_82f62a15469b470e85d6112c19a5cd64/
    Complete output (5 lines):
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-install-w7vliao8/madmom_82f62a15469b470e85d6112c19a5cd64/setup.py", line 11, in <module>
        from Cython.Distutils import build_ext
    ModuleNotFoundError: No module named 'Cython'
    ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/f6/38/64a07d5cabc3099484b177f4ad6c8b8148a11b6a21206a0729a257e43e97/madmom-0.13.tar.gz#sha256=4c5c01b2a1e106949c8d41533a3d4eccd0e57ab6529fea5ffe7b82b68caca270 (from https://pypi.org/simple/madmom/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
  Using cached madmom-0.12.1.tar.gz (3.8 MB)
    ERROR: Command errored out with exit status 1:
     command: ~/anaconda3/envs/ms9/bin/python -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-w7vliao8/madmom_ac180a0772974a78a608d37b5e14fcf3/setup.py'"'"'; __file__='"'"'/tmp/pip-install-w7vliao8/madmom_ac180a0772974a78a608d37b5e14fcf3/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-pip-egg-info-x3mnk4bj
         cwd: /tmp/pip-install-w7vliao8/madmom_ac180a0772974a78a608d37b5e14fcf3/
    Complete output (5 lines):
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-install-w7vliao8/madmom_ac180a0772974a78a608d37b5e14fcf3/setup.py", line 11, in <module>
        from Cython.Distutils import build_ext
    ModuleNotFoundError: No module named 'Cython'
    ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/c7/92/ec038532e1f174e0b9672662218b30faa4448c1438d4ba68b2289a7f8369/madmom-0.12.1.tar.gz#sha256=79b4fcffb3b1143e3ad80210cd095a5ff10c4f375d2f5451de69f68ad74410fc (from https://pypi.org/simple/madmom/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
  Using cached madmom-0.12.tar.gz (3.8 MB)
    ERROR: Command errored out with exit status 1:
     command: ~/anaconda3/envs/ms9/bin/python -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-w7vliao8/madmom_a4e37856317d4dff8a5c1de43a402dbe/setup.py'"'"'; __file__='"'"'/tmp/pip-install-w7vliao8/madmom_a4e37856317d4dff8a5c1de43a402dbe/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-pip-egg-info-rytzu2k5
         cwd: /tmp/pip-install-w7vliao8/madmom_a4e37856317d4dff8a5c1de43a402dbe/
    Complete output (5 lines):
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-install-w7vliao8/madmom_a4e37856317d4dff8a5c1de43a402dbe/setup.py", line 11, in <module>
        from Cython.Distutils import build_ext
    ModuleNotFoundError: No module named 'Cython'
    ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/28/0e/752fa1aa85c99a18036be8e2301cef9fc60a02824bba962052aa32c017a6/madmom-0.12.tar.gz#sha256=d21c7dd059c811bd8c5704c7fb18b3effa9f59d20224ca1cc6fae2cf70ae3df3 (from https://pypi.org/simple/madmom/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
ERROR: Could not find a version that satisfies the requirement madmom (from wavescapes) (from versions: 0.12, 0.12.1, 0.13, 0.13.1, 0.13.2, 0.14, 0.14.1, 0.15, 0.15.1, 0.16, 0.16.1)
ERROR: No matching distribution found for madmom
```

Let me know if you need me to provide further information.

**Comment by cedricviaccoz** [comment]
Thanks for the issue. I actually had that happening a while ago but could not replicate it, so I thought something went wrong with my local installation. This might require a full reupload to pip to properly fix unfortunately and unlikely I can tackle that before this week-end if not later next week. So if you really have to make wavescapes in the meantime, I know this issue does not arise on [google colaboratory](https://colab.research.google.com/) (basically the google docs of jupyter notebook)

**Comment by cedricviaccoz** [comment]
Another temporary fix might be to install Cython (`pip install Cython`), if you could try and let me know if that solve the issue for you, that will actually be sweet, I would only need to add it as an explicit dependancy in the pip package.

**Comment by johentsch** [comment]
Sweeeet, that _does_ fix the issue! However, just so are aware, wavescapes is currently not compatible with Python 3.10 because `madmom` isn't. I've [filed an issue](https://github.com/CPJKU/madmom/issues/502) with them.

**Comment by cedricviaccoz** [comment]
Hum interesting, thanks for noticing the incompatibility with 3.10. 

If madmom is hard to maintain as an out of the box feature of the library, I'll think about whether it should be made a "facultative" dependancy, meaning the "import madmom" only occurs when someone wants to do wavescapes on real audio. As I feel like the target audience of the library might be more enclined to use symbolic data rather than real audio, it would make the library a bit more easy to setup. 

On the flipside, people that _do_ want wavescapes on audio would have to set it up separately on their own if such decision is made

**Comment by johentsch** [comment]
At the same time, people analyzing audio might have their own preferred ways of getting PCVs, so having it as an optional depency is certainly an option. Currently, the interface is missing a way of easily passing your own PCVs (unless I overlooked it) and IMHO that functionality should be added anyway...

**Comment by fabianmoss** [comment]
Just had the same issue. Maybe it would be good to include a link to this issue to the README?

**Comment by cedricviaccoz** [comment]
Good suggestion Fabian, I changed the README to add information about this issue. The proper way to fix it will likely require to implement an optional import of madmom (or waiting on a stable version from them), and thus uploading a brand new version on the library on pip. Considering the amazing work Johannes is currently doing, I would like to package this fix with all the improvements he made in a 1.2 release. Closing this issue for now.
